### PR TITLE
Add account lockout

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: doc/integrator/authentication.rst [hunk 1]
```diff
@@ -42,3 +42,17 @@ To enable the two factors authentication you should set the following settings:
 
 If a user lost his second authentication factor he can't ask for a new one, to reset it the administrator
 should uncheck the 'The user changed his password' field on the user in the admin interface.
+
+Account lockout
+~~~~~~~~~~~~~~~
+
+To loc an account after a certain numer of failure set the following settings:
```
Comment (jwkaltz) [inline doc/integrator/authentication.rst]: To lock an account after a certain number of authentication failures, set the following:

File: doc/integrator/authentication.rst [hunk 2]
```diff
@@ -42,3 +42,17 @@ To enable the two factors authentication you should set the following settings:
...
+
+.. code:: yaml
+
+   vars:
+     authentication:
+       max_consecutive_failed: 10
+
+If a user will be locked, the administrator should uncheck the 'Deactivated' field on the user in the
```
Comment (jwkaltz) [inline doc/integrator/authentication.rst]: I am not sure what we are trying to say here. 

Is it: "To unlock a user, the administrator should ..."

File: geoportal/c2cgeoportal_geoportal/views/entry.py [hunk 3]
```diff
@@ -1265,6 +1277,9 @@ def loginresetpassword(self):  # pragma: no cover
         if error is not None:
             LOG.info(error)
             raise HTTPBadRequest("See server logs for details")
+        if user.deactivated:
+            LOG.info("The user '%s' is desactivated", username)
```
Comment (jwkaltz) [inline geoportal/c2cgeoportal_geoportal/views/entry.py]: deactivated

File: doc/integrator/authentication.rst (hunk 2 above)
Comment (sbrunner) [reply doc/integrator/authentication.rst]: Effectively thanks :-)

File: doc/integrator/authentication.rst [hunk 4]
```diff
@@ -42,3 +42,17 @@ To enable the two factors authentication you should set the following settings:
...
+Account lockout
+~~~~~~~~~~~~~~~
+
+To lock an account after a certain number of authentication failures, set the following settings:
+
+.. code:: yaml
+
+   vars:
+     authentication:
+       max_consecutive_failed: 10
```
Comment (arnaud-morvan) [inline doc/integrator/authentication.rst]: max_consecutive_failures

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/geoportal/CONST_config-schema.yaml [hunk 5]
```diff
@@ -178,6 +178,16 @@ mapping:
...
+      authentication:
+        type: map
+        required: false
+        mapping:
+          two_factor:
+            type: bool
+            required: false
+          max_consecutive_failed:
+            type: int
+            required: false
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/scaffolds/update/geoportal/CONST_config-schema.yaml]: Not really related with this PR but how about "two_factor_issuer_name" ? seems that two_factor_issuer_name is also missing in doc.

File: geoportal/c2cgeoportal_geoportal/views/entry.py [hunk 6]
```diff
@@ -1120,6 +1120,7 @@ def login(self):
                 if not user.validate_2fa_totp(self.request.POST.get("otp")):
                     raise HTTPUnauthorized("See server logs for details")
             user.update_last_login()
+            user.tech_data['consecutive_failed'] = 0
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/views/entry.py]: consecutive_failures

File: doc/integrator/authentication.rst (hunk 4 above)
Comment (sbrunner) [reply doc/integrator/authentication.rst]: +1 :-)

File: geoportal/c2cgeoportal_geoportal/scaffolds/update/geoportal/CONST_config-schema.yaml (hunk 5 above)
Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/scaffolds/update/geoportal/CONST_config-schema.yaml]: +1 :-)

**Pull request merged**